src/App.jsx

```diff
@@ -14,45 +15,94 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+export const App = () => {
+  const [sortedGoods, setSortedGoods] = useState([...goodsFromServer]);
+  const [isSorted, setIsSorted] = useState(false);
+  const [isReversed, setIsReversed] = useState(false);
+  const [sortButton, setSortButton] = useState(null);
+
+  const sortAlphabetically = () => {
+    const sorted = isReversed
+      ? [...sortedGoods]
+        .sort((good1, good2) => good1.localeCompare(good2))
+        .reverse()
+      : [...sortedGoods].sort((good1, good2) => good1.localeCompare(good2));
+
+    setSortedGoods(sorted);
+    setIsSorted(true);
+    setSortButton('alphabetically');
+  };
+
+  const sortByLength = () => {
+    const sorted = isReversed
+      ? [...sortedGoods].sort((good1, good2) => good2.length - good1.length)
+      : [...sortedGoods].sort((good1, good2) => good1.length - good2.length);
+
+    setSortedGoods(sorted);
+    setIsSorted(true);
+    setSortButton('byLength');
+  };
```

Comment:
Looks a bit complicated, instead you can create a function and pass a sortField param and inside this function you can do some manipulations based on sortField value (Alphabetically or byLength). This function should also accept a goods and return a new sortedGoods

Reply:
Yes, I replaced the two sort functions with one sortGoods.